modyn/models/resnet18/resnet18.py

```diff
@@ -1,10 +1,44 @@
 from typing import Any
 
-from torchvision import models
+import torch
+from modyn.models.modyn_model import ModynModel
+from torch import Tensor, nn
+from torchvision.models.resnet import BasicBlock, ResNet
 
 
 class ResNet18:
     # pylint: disable-next=unused-argument
     def __init__(self, model_configuration: dict[str, Any], device: str, amp: bool) -> None:
-        self.model = models.__dict__["resnet18"](**model_configuration)
+        self.model = ResNetModyn(model_configuration)
         self.model.to(device)
+
+
+# the following class is adapted from
+# torchvision https://github.com/pytorch/vision/blob/main/torchvision/models/resnet.py
+
+
+class ResNetModyn(ResNet, ModynModel):
+    def __init__(self, model_configuration: dict[str, Any]) -> None:
```

Comment:
Now it's more explicit since the context manager was difficult to use in the current codebase. You can call the method `model.embedding_recorder.start_recording()` to activate the recording and `model.embedding_recorder.end_recording()` to stop it. These methods will be called in the pytorch trainer during downsampling (`downsampling_batch` and `downsampling_trigger_training_set`) if the downsampling method requires `CoresetMethodsSupport`

Reply:
Sounds good and looks good! Thank you!